src/main/java/seedu/address/model/log/AppointmentDate.java

```diff
@@ -18,16 +22,19 @@ public class AppointmentDate {
      * @param date A valid LocalDate.
      */
     public AppointmentDate(LocalDate date) {
+        requireNonNull(date);
         this.date = date;
     }
 
     /**
      * Constructs an {@code AppointmentDate} by parsing a date string.
      *
-     * @param dateString A valid date string in the format of 'yyyy-MM-dd'.
+     * @param dateString A valid date string in the format of "dd MMM yyyy".
      */
     public AppointmentDate(String dateString) {
-        this.date = LocalDate.parse(dateString);
+        requireNonNull(dateString);
+        checkArgument(isValidDateString(dateString), MESSAGE_CONSTRAINTS);
+        this.date = LocalDate.parse(dateString, FORMATTER);
     }
```

Comment:
catches null date entry and returning message constraint, instead of throwing NULLEXCEPTION error. Check if Date is in correct format as required (dd MMM yyyy)